client/models/scenery_editor.js

```diff
@@ -318,16 +321,71 @@ class SceneryEditor {
             post('api/tools/scenery/instance/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeObject(document.getElementById('tools-detail-scenery-id').innerText);
             });
         } else if (this.selected_type === 'unique') {
             post('api/tools/scenery/unique/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeObject(document.getElementById('tools-detail-scenery-id').innerText);
             });
         }
     }
+
+    addObject(object){
+        let object_id = object.x + ',' + object.y;
+            
+        WORKSPACES.current_map.sceneryLoader.createScenery(object, (model, definition) => {
+            let m = createSceneryMesh(object_id, object, WORKSPACES.current_map.terrain, model, definition);
+            m.original_id = { type: 'scenery', id: object_id };
+            WORKSPACES.current_map.scenery_groups['trees'].add(m);
+            WORKSPACES.current_map.scenery_references[object_id] = {
+                instance: object,
+                definition: definition,
+                threeObject: m
+            }
+        })
+    }
+
+    editObject(object){
+        let newObject = WORKSPACES.current_map.scenery_references[object.id].instance;
+
+        if (object.modification == 'tint'){
```

Comment:
when you have things branches like this it usually means you should make different methods for these.

'if (foo) ....., if (bar) ...' usually means you should have a doFoo() and a doBar() method.